pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4295,202 +4416,122 @@ int BPF_KPROBE(cgroup_bpf_run_filter_skb)
+
     // inform userland about protocol family (for correct L3 header parsing)...
     struct sock_common *common = (void *) sk;
     u8 family = BPF_READ(common, skc_family);
     switch (family) {
-        case AF_INET:
+        case PF_INET:
             eventctx->retval |= family_ipv4;
+            l3_size = get_type_size(struct iphdr);
             break;
-        case AF_INET6:
+        case PF_INET6:
             eventctx->retval |= family_ipv6;
+            l3_size = get_type_size(struct ipv6hdr);
             break;
+        default:
+            return 1;
     }
     // ... through event ctx ret val
 
-    // set event arguments
-    neteventctx.bytes = 0; // no payload by default (changed inside skb prog)
-
-    // (*) Use skb timestamp as the key for a map shared between this kprobe and
-    // the skb ebpf program: this is **NOT SUPER** BUT, for older kernels, that
-    // provide ABSOLUTE NO eBPF helpers in cgroup/skb programs context, it does
-    // its job: pre-process everything HERE so cgroup/skb programs can use.
-    //
-    // Explanation: The cgroup/skb eBPF program is called right after this
-    //              kprobe, but preemption is enabled. If preemption wasn't
-    //              enabled, we could simply populate a single item map and pick
-    //              pointer inside cgroup/skb. Instead, we index map items
-    //              using the skb timestamp, which is a value that is shared
-    //              among this kprobe AND the cgroup/skb program context
-    //              (through its skbuf copy).
-    //
-    // Theoretically, map collisions might occur, BUT very unlikely due to:
-    //
-    // kprobe (map update) -> cgroup/skb (consume) -> kretprobe (map delete)
-
-    bpf_map_update_elem(&cgrpctxmap, &skbts, &neteventctx, BPF_NOEXIST);
-
-    return 0;
-}
-
-SEC("kretprobe/__cgroup_bpf_run_filter_skb")
-int BPF_KRETPROBE(ret_cgroup_bpf_run_filter_skb)
-{
-    // runs AFTER the CGROUP/SKB eBPF program
-
-    // pick from entry from entrymap
-    u32 host_tid = bpf_get_current_pid_tgid();
-    struct entry *entry = bpf_map_lookup_elem(&entrymap, &host_tid);
-    if (!entry) // no entry == no tracing
-        return 0;
-
-    // pick args from entry point's entry
-    // struct sock *sk = (void *) entry->args[0];
-    struct sk_buff *skb = (void *) entry->args[1];
-
-    // cleanup entrymap
-    bpf_map_delete_elem(&entrymap, &host_tid);
-
-    // use skb timestamp as the key for cgroup/skb
-    u64 skbts = BPF_READ(skb, tstamp);
-
-    // delete netctx after cgroup ebpf program runs
-    bpf_map_delete_elem(&cgrpctxmap, &skbts);
-
-    return 0;
-}
+    // read IP/IPv6 headers
 
-//
-// Type definitions and prototypes for protocol parsing
-//
-
-// NOTE: proto header structs need full type in vmlinux.h (for correct skb copy)
-
-typedef union iphdrs_t {
-    struct iphdr iphdr;
-    struct ipv6hdr ipv6hdr;
-} iphdrs;
+    void *data_ptr = NULL;
+    u16 mac_len = BPF_READ(skb, mac_len);
```

Comment:
split into 2 maps: ingress/egress netctx because the packet header would be the same for both (causing a conflict in the map). By having 2 cgrpctxmaps I can avoid that happening and still have the 2 different packets inside the map(s).